### PR TITLE
IJsonBsonConverterFactory. Incomplete - discussion requested

### PR DESCRIPTION
**Pull request by bboyle1234**
Inspired by https://github.com/OrleansContrib/Orleans.Providers.MongoDB/pull/50, 
I'm wondering if this is a solution that would help @sdghchj and others.

If the idea is approved, the following is still needed: 
1. Testing -- I need some help learning how to setup and run the tests.

### CODE REVIEW

File: Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs [hunk 1]
```diff
@@ -83,35 +69,78 @@ public static BsonValue ToBson(this JToken source)
                 case JTokenType.TimeSpan:
                     return BsonValue.Create(((JValue)source).ToString());
                 case JTokenType.Date:
```
Comment (SebastianStehle) [inline Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs]: I think you should introduce a virtual method for each case to make it easier to override it.

Comment (bboyle1234) [reply Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs]: Can do. That was my initial instinct, then after starting, I thought it was a bit over complicated. I will yield to your experience. Thank you.

Comment (bboyle1234) [reply Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs]: Here's a screenshot of partial changes using the concept described, if I understand correctly. I think this is what you mean ... I feel uncomfortable with it, how about you? 

image

Comment (bboyle1234) [reply Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs]: I've pushed the complete set of changes.

**Comment by SebastianStehle** [comment]
I feel very bad that I was not able to do something for the mongo provider. What is the status of this PR? Shall we try to get it in somehow or close it?

**Comment by SebastianStehle** [comment]
I have developed a new solution using readers and writers to make this more efficient: https://github.com/Squidex/squidex/blob/master/backend/src/Squidex.Infrastructure.MongoDb/MongoDb/BsonJsonReader.cs

**Pull request closed without merging**